File: app/typescript-cli/test_binding.ts
import imzxCore from '../bindings/typescript/src/index'; // Sesuaikan path jika perlu

async function runTest() {
    console.log("--- Starting TypeScript Binding Test ---");

    const name = "TS-Test-Agent";
    const description = "A test agent for TS";
    const prompt = "You are a test agent";

    console.log(`Initializing: ${name}`);
    // Note: In real Neon, agentNew might return a pointer/object
    const initMsg = imzxCore.agentNew(name, description, prompt);
    console.log(`Status: ${initMsg}`);

    const testInput = "Hello from TypeScript!";
    console.log(`Sending Input: '${testInput}'`);

    try {
        const response = imzxCore.agentRun(testInput);
        console.log(`Received Response: '${response}'`);

        // Validation
        if (response.includes("Rust-powered response") && response.includes(testInput)) {
            console.log("✅ SUCCESS: Agent responded correctly via Rust Core.");
        } else {
            console.error("❌ FAILED: Response content mismatch.");
            process.exit(1);
        }
    } catch (error) {
        console.error(`❌ FAILED: An error occurred: ${error}`);
        process.exit(1);
    }

    console.log("--- Test Completed Successfully ---");
}

runTest();

File: app/python-cli/test_binding.py
import imzx_core
import asyncio
import sys

async def test_agent_logic():
    print("--- Starting Python Binding Test ---")

    name = "TestAgent"
    description = "A test agent"
    prompt = "You are a test agent"

    print(f"Initializing: {name}")
    agent = imzx_core.PyAgent(name, description, prompt)

    test_input = "Hello, Rust!"
    print(f"Sending Input: '{test_input}'")

    # No try-except here to allow AssertionError to bubble up
    response = agent.run(test_input)
    print(f"Received Response: '{response}'")

    # Validation
    print("Validating response content...")
    assert "Hello! I received your message" in response
    assert test_input in response
    print("✅ SUCCESS: Agent responded correctly via Rust Core.")

    print("--- Test Completed Successfully ---")

if __name__ == "__main__":
    try:
        asyncio.run(test_agent_logic())
    except Exception as e:
        print(f"❌ CRITICAL ERROR: {e}")
        import traceback
        traceback.print_exc()
        sys.exit(1)
